packages/css/src/scales.ts

```diff
@@ -0,0 +1,332 @@
+import * as CSS from 'csstype'
+
+import { FinalTheme } from './types'
+
+type CollectionKeys<T> = T extends any[]
+  ? Exclude<keyof T, keyof any[]> | number
+  : keyof T
+
+// turn `string` to `string & {}` which can be part of a union
+// we use this to allow autocomplete on css globals
+type StringHack<T> = string extends T ? Exclude<T, string> | (string & {}) : T
+
+type ScaleProperty<TScale> =
+  | StringHack<CollectionKeys<Exclude<TScale, undefined>>>
+  | CSS.Globals
+
+const colors = {
+  color: 'colors',
+  backgroundColor: 'colors',
+  borderColor: 'colors',
+  caretColor: 'colors',
+
+  fill: 'colors',
+  stroke: 'colors',
+
+  borderTopColor: 'colors',
+  borderBottomColor: 'colors',
+  borderLeftColor: 'colors',
+  borderRightColor: 'colors',
+
+  outlineColor: 'colors',
+} as const
+
+export type Color = Exclude<ScaleProperty<FinalTheme['colors']>, 'modes'>
```

Comment:
What if we still included keys from the different color modes?

We could do something like this: [this](https://www.typescriptlang.org/play?ts=4.1.0-dev.20201005#code/MYewdgzgLgBKA2IBOEYF4YG8BQM8wAckBLAWwEMkBPALhgHIBzJAUxbHoBpd9SQATFhDo58YmP0oBrET3HiiZSrQZUW8RAHcuc+fgD0+mJoAW5WMQBmMcjD6CYZ1LYTIYUMxYgdY5AEYgAG4sAPy6eu4sAB5QdPQExGBS9OH4AL66GWk2qKCQUNjYUFQELDAAwiCISOjuJSwg1q4ohcWlMABC5BAsldWoGACiUcDwAK6CADxSLFSNFVXInAz2QvQAfK31MACyAr2LKLUzc9Z9yADa9KsQ9AC6Fyfz50hXN-d3W+0AYonk8C8Bp1ugd+jAAD67faAwp5aAwSx-eB0X5gf6Ai53WpXRQUahcBhQaJQe5AA).

Screenshot: https://i.imgur.com/L8BU0M5.png

Reply:
Isn't this a problem, though? If we use `"secondary"` in default color mode, we'll have `color: undefined` somewhere in our styles.

I don't have a strong opinion here. I always declare my colors like this anyway TBH

```tsx
const light = { ... };
type ColorMode = typeof light;
const dark: ColorMode = { ... };
const other: ColorMode = { ... };

const initialColorMode = "light";
const modes = { light, dark, other };

const colors = {
  ...modes[initialColorMode],
  modes: omit(modes, initialColorMode)
}
```

Reply:
I also don’t have a strong opinion on it. That’s probably the right way to do it, so we can stick without it.